internal/cache/lru/lru.go

```diff
@@ -267,3 +268,17 @@ func (c *Cache) UpdateWithoutChangingOrder(
 
 	return nil
 }
+
+func (c *Cache) EraseEntriesWithGivenPrefix(
+	prefix string) (value []ValueType) {
+	var deletedEntries []ValueType
+
+	for key := range c.index {
+		if strings.HasPrefix(key, prefix) {
+			e := c.Erase(key)
+			deletedEntries = append(deletedEntries, e)
+		}
+	}
+
+	return deletedEntries
```

Comment:
why do you need to returned deletedEntries if its not used>

Reply:
Erase function is returning deleted entry, so return it here as well. Although laterally found that Erase function entry is getting used other places . So Removed return from here. 

Done.